frappe/database/query.py

```diff
@@ -23,13 +26,49 @@
 TAB_PATTERN = re.compile("^tab")
 WORDS_PATTERN = re.compile(r"\w+")
 BRACKETS_PATTERN = re.compile(r"\(.*?\)|$")
-SQL_FUNCTIONS = tuple(f"{sql_function.value}(" for sql_function in SqlFunctions)  # ) <- ignore this comment.
 COMMA_PATTERN = re.compile(r",\s*(?![^()]*\))")
 
 # less restrictive version of frappe.core.doctype.doctype.doctype.START_WITH_LETTERS_PATTERN
 # to allow table names like __Auth
 TABLE_NAME_PATTERN = re.compile(r"^[\w -]*$", flags=re.ASCII)
 
+# Pattern to validate field names in SELECT:
+# Allows: name, `name`, name as alias, `name` as alias, `table name`.`name`, `table name`.`name` as alias, table.name, table.name as alias
+ALLOWED_FIELD_PATTERN = re.compile(r"^(?:`?[\w\s-]+`?\.)?(`?\w+`?|\w+)(?:\s+as\s+\w+)?$", flags=re.ASCII)
```

Comment:
I strongly feel, if we want to do this right, we should drop all kinds of regex based validations.  The maxim for security should be "_parse_, don't validate". Parse known supported values and convert them to typical code that you'd write by hand. Don't bypass ANY QB parameterization. 

e.g. `selec table.col as projection` should become `frappe.qb.from(table).select(table.col).as_(projection`.

Reply:
Checked some more code. Looks like there is both parsing and validation :smile:

Reply:
Ig the design is mostly fine, we can get rid of risky things like functions and operators? Those should be implemented in code IMO.